tests/server/test_upgrade.py

```diff
@@ -339,6 +339,19 @@ def status_code = http.remove("status_code");
     }
 }
 
+// per https://github.com/elastic/apm/issues/21
+//  if kubernetes.node.name is set, copy it to host.hostname
+//  else if other kubernetes.* is set, remove host.hostname
+//  else leave it alone
+if (ctx._source.kubernetes?.node?.name != null) {
+    if (! ctx._source.containsKey("host")) {
+        ctx._source.host = new HashMap();
+    }
+    ctx._source.host.hostname = ctx._source.kubernetes.node.name;
+} else if (ctx._source.containsKey("kubernetes")) {
+    ctx._source.host.remove("hostname");
```

Comment:
This is not really tested, as none of the 6.x documents has a `host.hostname`.

Reply:
Good call.  This happens after the `system` -> `host` move, I'll add a comment to make that dependency clear.  I'll also add a guard since that can be null also.

Reply:
Ah right, I needed to look for the `system.hostname` instead of `host.hostname`.